src/hooks/useRefCallback/useRefCallback.ts

```diff
@@ -5,28 +5,33 @@ type Nullable<T> = T | null
 type RefCallback<T> = {
   onInit: (node: Nullable<T>) => void
   onDestroy?: (node: Nullable<T>) => void
+  deps?: Array<any> // Any dependenies to create a new callback reference
```

Comment:
We should be able to use the same with a `useEffect()` in the caller, will test it out and update

Reply:
Haven't managed to find a better solution for that other than another hook on top of `useRefCallback`, which is probably overkill for this. Since `useRefCallback()` is used only for `<AssetList />` and `deps` is optional, looking back into this implementation, it's probably fine!

Reply:
Yep, I think this is fine for now. It all works - we can refactor later if we really need to.